Events.php

```diff
@@ -17,12 +17,23 @@ public static function onAuthClientCollectionInit($event)
         /** @var Collection $authClientCollection */
         $authClientCollection = $event->sender;
 
+        $tenantId = ConfigureForm::getInstance()->directoryId;
+        if($tenantId){
+            $authUrl = "https://login.microsoftonline.com/".(ConfigureForm::getInstance()->directoryId)."/oauth2/v2.0/authorize";
```

Comment:
I don't think we should do this ideally within the `Events.php` and instead should implement an additional `$mode` variable within the configuration settings model;

Something like the following?

```php
    const DIRECTORY_MODE = 'https://login.microsoftonline.com/' . self::getInstance()->directoryId . '/oauth2/v2.0/authorize';
    const COMMON_MODE = 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize';

    public $mode;

    public static function getMode($mode) : array
    {
        return self::getModes()[$mode];
    }

    /**
     * @return requests for Directory ID/Common modes
     */
    public static function getModes($selectable = true) : array
    {
        $mode = [
            self::DIRECTORY_MODE => 'Directory ID',
            self::COMMON_MODE => 'Common',
        ];

        if ($selectable) {
            return $mode;
        }

        return $mode;
    }
```

Then in the configuration settings view we add the following;
```php
<?= $form->field($model, 'mode')->dropDownList($model->getModes(true)); ?>
```

We can also add variables for  `$authUrl` and `$tokenUrl` which we then later introduce within the `MicrosoftAuth` class something like the following?
https://github.com/GreenMeteor/auth-mastodon/blob/main/authclient/Mastodon.php#L49-L62